src/serializers/fields.rs

```diff
@@ -210,7 +211,28 @@ impl GeneralFieldsSerializer {
             // Check for missing fields, we can't have extra fields here
             && self.required_fields > used_req_fields
         {
-            Err(PydanticSerializationUnexpectedValue::new_err(None))
+            let required_fields = self.required_fields;
+            let field_name = extra.field_name.unwrap_or("<unknown field>").to_string();
+            let type_name = match extra.model {
+                Some(model) => model
+                    .get_type()
+                    .qualname()
+                    .ok()
+                    .unwrap_or_else(|| PyString::new_bound(py, "<unknown python object>"))
+                    .to_string(),
+                None => "<unknown python object>".to_string(),
+            };
+
+            let field_value = match extra.model {
+                Some(model) => truncate_safe_repr(model, Some(100)),
+                None => "<unknown python object>".to_string(),
+            };
+
+            Err(PydanticSerializationUnexpectedValue::new_err(
+                Some(format!(
+                    "Expected {required_fields} fields but got {used_req_fields} for field {field_name} of type `{type_name}` with value `{field_value}` - serialized value may not be as expected."
```

Comment:
The use of `for field {field_name}` is confusing to me here because every other part of the error is at the model level, i.e. this model has 4 fields but we only got 2 fields, this was the model data.

The field, on the other hand, is, if I understand it, the name of the parent model that this model has been found at. Which is useful debug info perhaps, as long as we make that clearer in the wording.